From da87f91ad8450ccc5274cd7b6ba8d823b396c96f Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 18 Jun 2024 15:33:30 +0100
Subject: [PATCH 1138/1145] drm/bridge: panel: Ensure backlight is reachable

Ensure that the various options of modules vs builtin results
in being able to call into the backlight code.

https://github.com/raspberrypi/linux/issues/6198

Fixes: 573f8fd0abf1 ("drm/bridge: panel: Name an associated backlight device")
Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/bridge/panel.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/gpu/drm/bridge/panel.c
+++ b/drivers/gpu/drm/bridge/panel.c
@@ -87,8 +87,10 @@ static int panel_bridge_attach(struct dr
 	drm_connector_attach_encoder(&panel_bridge->connector,
 					  bridge->encoder);
 
+#if IS_REACHABLE(CONFIG_BACKLIGHT_CLASS_DEVICE)
 	backlight_set_display_name(panel_bridge->panel->backlight,
 				   panel_bridge->connector.name);
+#endif
 
 	if (bridge->dev->registered) {
 		if (connector->funcs->reset)
